core/schemas/user.py

```diff
@@ -16,10 +19,36 @@
     "reader": roles.Role.READER,
     "writer": roles.Role.WRITER,
 }
+SECRET_KEY = yeti_config.get("auth", "secret_key")
+ALGORITHM = yeti_config.get("auth", "algorithm")
+
+
+def create_access_token(
+    data: dict, expires_delta: datetime.timedelta | None = None
+) -> str:
+    to_encode = data.copy()
+    expire = None
+    if expires_delta:
+        expire = datetime.datetime.now(datetime.timezone.utc) + expires_delta
+    to_encode.update({"exp": expire})
+    encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
+    return encoded_jwt
+
+
+class RegisteredApiKey(BaseModel):
+    name: str
+    sub: str
+    scopes: list[str]
+    created: datetime.datetime = Field(default_factory=now)
+    exp: datetime.datetime | None = None
+    last_used: datetime.datetime | None = None
+    enabled: bool = True
 
-
-def generate_api_key():
-    return secrets.token_hex(32)
+    @property
```

Comment:
We should also add `@computed_field` to validate the [schema when serializing](https://docs.pydantic.dev/2.10/concepts/fields/#the-computed_field-decorator).